File: data-service/spring-data-provider/src/main/java/org/aykhan/dataprovider/dto/user/UserRequest.java
package org.aykhan.dataprovider.dto.user;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
public class UserRequest implements Serializable {
  private String name;
  private String surname;
  private String email;
  private String nickname;
  private String profilePicture;
}

File: data-service/scala-sql-dataprovider/app/dtos/user/UserRequest.scala
package dtos.user

import play.api.libs.json.Json

case class UserRequest(nickname: String,
                       name: Option[String],
                       surname: Option[String],
                       profilePicture: Option[String],
                       email: Option[String])

object UserRequest{
  implicit val reads = Json.reads[UserRequest]
}